fdbserver/Status.actor.cpp

```diff
@@ -1151,26 +1151,61 @@ ACTOR static Future<Void> consistencyCheckStatusFetcher(Database cx, JsonBuilder
 	return Void();
 }
 
+struct LogRangeAndUID {
+	KeyRange range;
+	UID destID;
+
+	LogRangeAndUID(KeyRange const& range, UID const& destID) : range(range), destID(destID) {}
+
+	bool operator < (LogRangeAndUID const& r) const { 
+		if(range.begin != r.range.begin) return range.begin < r.range.begin;
+		if(range.end != r.range.end) return range.end < r.range.end;
+		return destID < r.destID; 
+	}
+};
+
 ACTOR static Future<Void> logRangeWarningFetcher(Database cx, JsonBuilderArray *messages, std::set<std::string> *incomplete_reasons) {
 	try {
 		state Transaction tr(cx);
+		state Future<Void> timeoutFuture = delay(5.0);
```

Comment:
If we are using this in multiple places as my other suggestion does, it may be easier to do something like the following rather than check for readiness each time:
```suggestion
		state Future<Void> timeoutFuture = timeoutError(Never(), 5.0);
```